Framework/HistogramData/test/FixedLengthVectorTest.h

```diff
@@ -167,25 +166,28 @@ class FixedLengthVectorTest : public CxxTest::TestSuite {
     TS_ASSERT_EQUALS(dest[1], 0.1);
   }
 
-  void test_copy_assignment_fail() {
```

Comment:
Since there are tests here to make sure the copy and move assignment operators fails I assume that this is the desired behaviour. Does it make sense to change this, is there away to fix the warning without changing this behaviour?

Reply:
I added a copy assignment operator, since the copy constructor does the exact same thing. Basically this allowed at the moment:

```
Foo a = a;
Foo b (a); // Makes a copy fine

Foo c = a; // Fails
Foo c (a); // Will work for you even though it is the same as above
```
Apart from some extremely niche cases where self assignment caused invalid memory in legacy code which couldn't be changed this should do the same thing.

As to the why, I'm not sure....maybe someone once ran into a bug with an incorrect copy assignment operator (as the assignment didn't use copy-and-swap) and went on a crusade to remove it from multiple of our classes. This isn't the only one where I've re-added the ability.

Reply:
Ah okay thanks for clarifying, I just wanted to be sure given the test was added for this one in particular.
(also sorry for taking so long to follow up on this)